### ISSUE TITLE
Ocaml LSP causes over 90% cpu usage

### ISSUE THREAD

**Issue by Akin909** [neoclide/coc.nvim#244]
**Result from CocInfo**
Run `:CocInfo` command and paste the content below.
## versions

vim version: NVIM v0.3.1
node version: v10.3.0
coc.nvim version: 0.0.37
term: tmux-256color
platform: linux

## Error messages

**Describe the bug**
When editing a reasonml file using the `ocaml-language-server` nvim uses over 90% of my cpu

I've previously been using this same lsp with the same config with `autozimu/LanguageClient` before I discovered this awesome project :heart_eyes:, but I did not have the same issue there

I've tried setting my `coc_node_path` to a node version@10.3.0 and running `yarn` in the `coc` directory to make sure it loads v10.3.0 but that doesn't help :sob: 

**To Reproduce**

1. Open a `reasonml` file and start editing
2. check cpu usage using activity monitor or `htop`

**Screenshots**
![cpu_usage_coc](https://user-images.githubusercontent.com/22454918/49685968-5d9fab00-fae5-11e8-95c9-5b3409ec21d8.png)

**Comment by chemzqm** [comment]
Could be a problem of wrong root path, try set `"rootPatterns"` in `"ocaml"` section, and checkout the options send to server by use output channel: https://github.com/neoclide/coc.nvim/wiki/Debug-language-server#using-output-channel

**Comment by Akin909** [comment]
@chemzqm thanks for the quick response :+1:, I tried setting `['dune', 'dune-project', .git/]` as the `rootMarkers` and the output of `workspace.showOutput` show the root path as the correct root dir for my project :confused: 

I've tried with a typescript file and I dont get the same cpu usage issues, so I think it's probably something to do with the config for ocaml not sure what though :sob:

**Comment by chemzqm** [comment]
There's no `rootMarkers`, it's `rootPatterns `.

coc-tsserver doen't need rootPath.

**Comment by Akin909** [comment]
Sorry I meant `rootPatterns` I just typed the wrong thing in my comment, I also just tried upgrading and I'm still seeing really high CPU usage but no errors in `workspace.showOutup` or `CocErrors`.

**Comment by oblitum** [comment]
@Akin909 I was getting this with pyls, but their issue tracker also presents these kind of issues: https://github.com/neoclide/coc-rls/issues/5#issuecomment-445261217.

Gladly it has stopped. `bash-language-server` workspace script lookup also got fixed lately.

Are you using Ocaml or Reason? I've put setup for another Reason language server on Wiki, and edited your contribution there. The Reason language server doesn't present this issue for me.

**Comment by oblitum** [comment]
Notice though you shouldn't put both servers at same time to handle Reason filetypes.

**Comment by Akin909** [comment]
Unfortunately I started with the `reason-language-server` but it also throws error (not CPU errors, permissions errors, not related to this plugin) and I think isn't recommend for native reason project (rather than bucklescript ones)

**Comment by oblitum** [comment]
@Akin909 yep, reason-language-server not for native, just BuckleScript, but it's working fine, for me at least.

**Comment by oblitum** [comment]
@Akin909 which plugin?

**Comment by oblitum** [comment]
@Akin909 there was just another case incompatibility here https://github.com/neoclide/coc.nvim/issues/245

**Comment by Akin909** [comment]
@oblitum I'm slightly guessing as I tried a lot of things,  I basically
* deleted and re-installed the language server
* deleted and re-installed opam, merlin
* commented out most of my plugins and seems https://github.com/lambdalisue/gina.vim - the status item was causing quite high cpu usage, I've had it active for a week and didn't notice it so not sure if theres anyway it might interact with an lsp :man_shrugging: or if its really the cause tbh but I've been editing with `coc` (without the status item) for an hour and a bit and cpu usage is tiny

**Comment by oblitum** [comment]
> @Akin909 yep, reason-language-server not for native, just BuckleScript, but it's working fine, for me at least.

I withdraw that, the reason-language-server is working fine for native projects, I'm trying out with a basic esy based project, it's working fine.
